lib/mayaUsd/utils/layerLocking.cpp

```diff
@@ -17,12 +17,67 @@
 #include "layerLocking.h"
 
 #include <mayaUsd/listeners/notice.h>
+#include <mayaUsd/ufe/Global.h>
+#include <mayaUsd/ufe/Utils.h>
+#include <mayaUsd/utils/query.h>
+#include <mayaUsd/utils/util.h>
 
 #include <pxr/base/tf/weakBase.h>
 
+#include <ufe/path.h>
+
 namespace MAYAUSD_NS_DEF {
 
-namespace {
+MStatus copyLayerLockingToAttribute(MayaUsdProxyShapeBase* proxyShape)
+{
+    if (proxyShape != nullptr) {
+        LockedLayers& lockedLayers = getLockedLayers();
+
+        std::vector<std::string> toAttribute;
+        toAttribute.reserve(lockedLayers.size());
+
+        for (auto layer : lockedLayers) {
+            toAttribute.emplace_back(layer->GetIdentifier());
+        }
+        return proxyShape->setLockedLayers(toAttribute);
+    } else {
+        return MStatus::kFailure;
```

Comment:
IDK if it should be considered failure, but as long as we don't end-up printing error in the log, it's fine.